cloud_tasks/http_queues.py

```diff
@@ -0,0 +1,229 @@
+    parsedUri = urllib.parse.urlparse(uri)
+
+    http_target = {
+        "uri_override": {
+            "host": parsedUri.hostname,
+            "uri_override_enforce_mode": 2,  # ALWAYS use this endpoint
+        }
+    }
+    if parsedUri.scheme == "http":  # defaults to https
+        http_target["uri_override"]["scheme"] = 1
+    if parsedUri.port:
+        http_target["uri_override"]["port"] = f"{parsedUri.port}"
+    if parsedUri.path:
+        http_target["uri_override"]["path_override"] = {"path": parsedUri.path}
+    if parsedUri.query:
+        http_target["uri_override"]["query_override"] = {
+            "query_params": parsedUri.query
+        }
+
+    # Use the client to send a CreateQueueRequest.
+    queue = client.create_queue(
+        tasks.CreateQueueRequest(
+            parent=client.common_location_path(project, location),
+            queue={
+                "name": f"projects/{project}/locations/{location}/queues/{name}",
+                "http_target": http_target,
+            },
+        )
+    )
+
+    return queue
+
+
+# [END cloud_tasks_create_http_queue]
+
+
+# [START cloud_tasks_update_http_queue]
+import google.protobuf
+from google.cloud import tasks_v2beta3 as tasks
+
+
+def update_http_queue(
+    queue: tasks.Queue,
+    uri: str = "",
+    max_per_second: float = 0.0,
+    max_burst: int = 0,
+    max_concurrent: int = 0,
+    max_attempts: int = 0,
+) -> tasks.Queue:
+    """Update an HTTP queue with provided properties.
+    Args:
+        queue: The queue to update.
+        uri: The new HTTP endpoint
+        max_per_second: the new maximum number of dispatches per second
+        max_burst: the new maximum burst size
+        max_concurrent: the new maximum number of concurrent dispatches
+        max_attempts: the new maximum number of retries attempted
+    Returns:
+        The updated queue.
+    """
+
+    # Create a client.
+    client = tasks.CloudTasksClient()
+
+    update_mask = google.protobuf.field_mask_pb2.FieldMask(
+        paths=[]
+    )  # Track which fields need to be updated
+
+    if uri:
+        parsedUri = urllib.parse.urlparse(uri)
+
+        http_target = {
+            "uri_override": {
+                "host": parsedUri.hostname,
+                "uri_override_enforce_mode": 2,  # ALWAYS use this endpoint
+            }
+        }
+        if parsedUri.scheme == "http":  # defaults to https
+            http_target["uri_override"]["scheme"] = 1
+        if parsedUri.port:
+            http_target["uri_override"]["port"] = f"{parsedUri.port}"
+        if parsedUri.path:
+            http_target["uri_override"]["path_override"] = {"path": parsedUri.path}
+        if parsedUri.query:
+            http_target["uri_override"]["query_override"] = {
+                "query_params": parsedUri.query
+            }
+
+        queue.http_target = http_target
+        update_mask.paths.append("http_target")
+
+    if max_per_second != 0.0:
```

Comment:
Since these values are in the default variables, is there any reason why these have to be conditional assignments? Is there a reason why max_burst can't be send as it's default of 0, for example?

Reply:
I think at least one in the API treats 0 slightly different from the default.

Reply:
Looking at it again, the real reason is that if the default value is given, we don't want to change the queue's existing value, which may not be zero.